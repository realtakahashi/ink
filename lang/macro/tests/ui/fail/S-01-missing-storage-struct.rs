use ink_lang as ink;

#[ink::contract(version = "0.1.0")]
mod missing_storage_struct {
    // We are missing the #[ink(storage)] attribute here
    struct MissingStorageStruct {}

    impl MissingStorageStruct {
        #[ink(constructor)]
        fn constructor() -> Self {}

        #[ink(message)]
        fn message(&self) {}
    }
}

fn main() {}
