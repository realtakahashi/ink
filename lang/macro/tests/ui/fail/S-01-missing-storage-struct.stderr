error: no #[ink(storage)] struct found but expected exactly 1
 --> $DIR/S-01-missing-storage-struct.rs:3:1
  |
3 | #[ink::contract(version = "0.1.0")]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in an attribute macro (in Nightly builds, run with -Z macro-backtrace for more info)
